app/scripts/modules/core/state.service.ts

```diff
@@ -1,8 +1,27 @@
-import {IStateParamsService, IStateService} from 'angular-ui-router';
+import { module, IRootScopeService } from 'angular';
+import { IStateParamsService, IStateService } from 'angular-ui-router';
+import { Subject } from 'rxjs/Subject';
 
-export let stateService: IStateService = undefined;
-export let stateParamsService: IStateParamsService = undefined;
+export class StateService {
```

Comment:
Very good points. After talking offline, we agreed to split StateService into two things, `uirouter.ts` which is effectively the ngimport of ui-router, and `state.events` to encapsulate the rx-ing of state events.